Dockerfile

```diff
@@ -205,6 +205,22 @@ RUN if [ ! -z "${PG_LOGERRORS}" ]; then \
             && make clean && PG_CONFIG=/usr/lib/postgresql/${pg}/bin/pg_config make install || exit 1 ; \
         done; \
     fi
+
+# Allow Adding Extensions allows *new* files to be created, so that extensions can be added to a running container.
+# Existing files are still owned by root and have their sticky bit (the 1 in the 1750 permission mode) set,
+# and therefore cannot be overwritten or removed by the unprivileged (postgres) user.
+# This ensures the following:
+# - libraries and supporting files for extensions that are part of this Docker Image are immutable
+# - new libraries/files can be added to the container and updated by the postgres user
+ARG ALLOW_ADDING_EXTENSIONS=
+RUN if [ "${ALLOW_ADDING_EXTENSIONS}" = "true" ]; then \
+        for pg in ${PG_VERSIONS}; do \
+            for dir in "$(/usr/lib/postgresql/${pg}/bin/pg_config --sharedir)/extension" "$(/usr/lib/postgresql/${pg}/bin/pg_config --pkglibdir)"; do \
```

Comment:
I wonder if we need to add bindir here as well; some extensions, i.e. pg_repack, ship binaries in addition to the libraries and sql files.

Reply:
I'd be happy to add those once we run into those issues, although for binaries I think other mechanisms may be better.